common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java

```diff
@@ -300,6 +302,17 @@ protected void renewAccessToken(@NonNull final SilentTokenCommandParameters para
 
             // Set the AuthenticationResult on the final result object
             acquireTokenSilentResult.setLocalAuthenticationResult(authenticationResult);
+        } else {
+            if (tokenResult.getErrorResponse() != null) {
+                final String errorCode = tokenResult.getErrorResponse().getError();
+                final String subErrorCode = tokenResult.getErrorResponse().getSubError();
+                Logger.info(TAG, "Error: " + errorCode + " Suberror: " + subErrorCode);
+
+                if (errorCode.equals(INVALID_GRANT) && subErrorCode.equals(BAD_TOKEN)) {
+                    Logger.info(TAG, "Refresh token is invalid, deleting from cache");
+                    tokenCache.removeCredential(cacheRecord.getRefreshToken());
+                }
+            }
```

Comment:
Can we add one more `else` case here where if the request isn't successful, but _also_ doesn't contain an error response, we log a warning message?

Reply:
(Obviously, this condition _shouldn't_ ever happen)

Reply:
yep, should never happen, but no harm adding